content/es/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch.md

```diff
@@ -0,0 +1,536 @@
+por lo tanto strategic merge patch utiliza la estrategia de patch de default, la cual es `replace`.
+
+```go
+type PodSpec struct {
+  ...
+  Tolerations []Toleration `json:"tolerations,omitempty" protobuf:"bytes,22,opt,name=tolerations"`
+  ...
+}
+```
+
+## Usa JSON merge patch para actualizar un Deployment
+
+Un strategic merge patch es distinto a un 
+[JSON merge patch](https://tools.ietf.org/html/rfc7386).
+
+Con JSON merge patch, si quieres actualizar una lista
+tienes que especificar la lista nueva en su totalidad
+y reemplazar la lista existente con la lista nueva.
+
+el comando `kubectl patch` tiene un parámetro `type` que acepta los siguientes valores:
+
+<table>
+  <tr><th>Valor del parámetro</th><th>tipo de unión</th></tr>
+  <tr><td>json</td><td><a href="https://tools.ietf.org/html/rfc6902">JSON Patch, RFC 6902</a></td></tr>
+  <tr><td>merge</td><td><a href="https://tools.ietf.org/html/rfc7386">JSON Merge Patch, RFC 7386</a></td></tr>
+  <tr><td>strategic</td><td>Strategic merge patch</td></tr>
+</table>
+
+Para una comparación entre JSON patch y JSON merge patch, revisa
+[JSON Patch y JSON Merge Patch](https://erosb.github.io/post/json-patch-vs-merge-patch/).
+
+El valor predeterminado para el vparametro `type` es  `strategic`. entonces en el ejercicio
+anterior hiciste un strategic merge patch.
+
+A continuación has un JSON merge path en el mismo Deployment. crea un archivo llamado
+`patch-file-2.yaml` que tenga el siguiente contenido:
+
+```yaml
+spec:
+  template:
+    spec:
+      containers:
+      - name: patch-demo-ctr-3
+        image: gcr.io/google-samples/node-hello:1.0
+```
+
+en el comando patch configura el valor de `type` como `merge`
+
+```shell
+kubectl patch deployment patch-demo --type merge --patch-file patch-file-2.yaml
+```
+
+Revisa el Deployment modificado:
+
+```shell
+kubectl get deployment patch-demo --output yaml
+```
+
+la lista `containers` que especificaste en el patch solo tiene un contenedor.
+el resultado muestra que tu lista con un contenedor reemplazo a la lista `containers` preexistente.
+
+```yaml
+spec:
+  containers:
+  - image: gcr.io/google-samples/node-hello:1.0
+    ...
+    name: patch-demo-ctr-3
+```
+revisa los Pods en ejecución:
+
+```shell
+kubectl get pods
+```
+
+En el resultado se puede ver que los Pods existentes fueron temrinados y se
+crearon Pods nuevos. el `1/1` indica que cada Pod nuevo esta ejecutando un solo
+contenedor.
+
+```shell
+NAME                          READY     STATUS    RESTARTS   AGE
+patch-demo-1307768864-69308   1/1       Running   0          1m
+patch-demo-1307768864-c86dc   1/1       Running   0          1m
+```
+
+## Usa strategic merge patch para actualizar un Deployment utilizando la estrategia retainKeys  
+
+Aquí esta el archivo de configuración para un deployment que usa la estrategia `RollingUpdate`:
+
+{{% code_sample file="application/deployment-retainkeys.yaml" %}}
```

Comment:
Similarly, create new file for Spanish by copying the `deployment-retainkeys.yaml` file under [content/en/examples/application/deployment-retainkeys.yaml](https://github.com/kubernetes/website/blob/main/content/en/examples/application/deployment-retainkeys.yaml) to a similar location for Spanish by creating the "deployment-retainkeys.yaml" file under "[content/es/examples/application/](https://github.com/kubernetes/website/blob/main/content/es/examples/application/)".